cinch/playbooks/jenkins_restart_utility.yml

```diff
@@ -0,0 +1,139 @@
+# This playbook aims to gather diagnostics data in the event of
+# trouble with Jenkins, and then restarts Jenkins to keep operations running.
+# By default, only the diagnostics data is collected, but the Jenkins service
+# restart can be enabled by setting the 'restart_jenkins' variable to true:
+
+# Example:
+# ansible-playbook -u root --private-key /path/to/key
+# -i '10.0.183.169', jenkins_restart_utility.yml -e "restart_jenkins=true"
+
+- hosts: all
+  vars:
+    jenkins_pid_path: '/run/jenkins.pid'
+  tasks:
+    # This block ignores all errors to ensure that Jenkins is restarted on
+    # demand even if we can't gather diagnostics data.
+    - name: gather diagnostics data
+      block:
+        - name: check if Jenkins is running
+          stat:
```

Comment:
This was a suggestion by @olivergondza, and if possible I'd like to get his thoughts on this.  It may be the case that the diagnostics collection that depends on the Jenkins PID would be unavailable anyways in the case of a Jenkins crash.